frame/balances/src/impl_currency.rs

```diff
@@ -487,7 +487,9 @@ where
 			return true
 		}
 		Self::account(who).free.checked_sub(&value).map_or(false, |new_balance| {
-			Self::ensure_can_withdraw(who, value, WithdrawReasons::RESERVE, new_balance).is_ok()
+			new_balance >= T::ExistentialDeposit::get() &&
```

Comment:
Do we still need `can_` functions when we are using storage-layers (#12012)?

Reply:
It depends. `Currency` trait functions don't necessarily get called from within a storage transaction. (XCM uses them for example.)